plutus-core/plutus-ir/src/PlutusIR/Transform/Inline.hs

```diff
@@ -102,13 +102,14 @@ type ExternalConstraints tyname name uni fun m =
     , MonadQuote m
     )
 
-type Inlining tyname name uni fun a m =
-    ( MonadState (Subst tyname name uni fun a) m
-    , MonadReader Deps.StrictnessMap m
-    , MonadQuote m
-    , ExternalConstraints tyname name uni fun m
+type InliningConstraints tyname name uni fun =
+    ( HasUnique name TermUnique
+    , HasUnique tyname TypeUnique
+    , PLC.ToBuiltinMeaning uni fun
     )
 
+type Inlining tyname name uni fun a = ReaderT Deps.StrictnessMap (StateT (Subst tyname name uni fun a) Quote)
```

Comment:
Usually we call our monad types `SomethingM`, maybe `InlineM`?

Reply:
Also a comment: "Using a concrete monad makes a very large difference to the performance of this module (determined from profiling)"